framework/src/main/java/org/checkerframework/framework/util/typeinference8/util/Resolution.java

```diff
@@ -213,36 +213,65 @@ private BoundSet resolveSmallestSet(Set<Variable> as, BoundSet boundSet) {
   }
 
   /**
-   * Resolves all the non-capture variables in {@code variables}.
+   * Resolves all the non-capture variables in {@code vars}.
    *
-   * @param variables the variables
+   * @param vars the variables
    */
-  private void resolveNonCapturesFirst(List<Variable> variables) {
-    Variable smallV;
-    do {
-      smallV = null;
+  private void resolveNonCapturesFirst(List<Variable> vars) {
+    // Variables that are not captures and need to be resolved. (Avoid side-effecting vars)
+    List<Variable> variables = new ArrayList<>(vars);
```

Comment:
Please don't use `variables` as the variable name.  Use `toResolve` or `nonCaptures` or some other term that conveys what these variables are.